backend/director/agents/editing/code_executor.py

```
@@ -0,0 +1,190 @@
+
+class CodeExecutor:
+    """Executes Timeline Python code safely."""
+
+    def __init__(self, session):
+        self.session = session
+
+    def execute_code(self, code: str, description: str = "") -> AgentResponse:
+        try:
+            conn = self.session.state.get("conn")
+            if not conn:
+                raise ValueError("No VideoDB connection available in session")
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)